### PR TITLE
Migrate to latest lints, bump min SDK to Dart 3.2

### PR DESCRIPTION
**Pull request by kevmoo**

### CODE REVIEW

File: lib/html.dart [hunk 1]
```diff
@@ -102,7 +102,9 @@ class HtmlWebSocketChannel extends StreamChannelMixin
       // Unfortunately, the underlying WebSocket API doesn't expose any
       // specific information about the error itself.
       final error = WebSocketChannelException('WebSocket connection failed.');
-      _readyCompleter.completeError(error);
+      if (!_readyCompleter.isCompleted) {
+        _readyCompleter.completeError(error);
```
Comment (kevmoo) [inline lib/html.dart:106]: FYI! A test was failing without this...very weird!

Comment (natebosch) [reply lib/html.dart:106]: That could indicate a bug in the changes...

Comment (natebosch) [reply lib/html.dart:106]: I also can't repro this. Do you know which test it was?

Comment (kevmoo) [reply lib/html.dart:106]: html_test – running with latest Flutter dev release

Comment (natebosch) [reply lib/html.dart:106]: Ah, I don't know why I overlooked running on chrome...

**Review by natebosch — APPROVED** [review]

**Pull request merged**